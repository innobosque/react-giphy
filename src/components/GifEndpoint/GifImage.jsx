import React from 'react';
import Proptypes from 'prop-types';

export const GifImage = ({title,url}) => {
  return (
    <img className="w-full h-full object-cover" src={url} alt={title} />
  )
}

GifImage.propTypes = {
    title: Proptypes.string,
    url: Proptypes.string.isRequired
}

GifImage.defaultProps = {
    title: ''
}
