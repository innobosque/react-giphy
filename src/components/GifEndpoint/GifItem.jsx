import { LoadingSpinner } from "../LoadingSpinner";
import { useFetchGifs } from "../../hooks/useFetchGifs";
export const GifItem = ({search} ) => {
    const {id,category} = search;
    const {isLoading,error,images} = useFetchGifs(category);
    return (
        <>
            {isLoading && <LoadingSpinner widthHeight={'40'} />}
            {!!error.message && error.message}
            {images.map(image=>(
                <img className="w-full h-full object-cover" key={image.id} src={image.url} alt={image.title} title={image.title} />
            ))}
        </>
    )
}