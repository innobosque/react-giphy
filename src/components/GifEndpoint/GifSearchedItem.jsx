import { LoadingSpinner } from "../LoadingSpinner";
import { useFetchGifs } from "../../hooks/useFetchGifs";
import { GifImage } from "./GifImage";
export const GifSearchedItem = ({search} ) => {
    const {id,category} = search;
    const {isLoading,error,images} = useFetchGifs(category);
    return (
        <>
            {isLoading && <LoadingSpinner widthHeight={'40'} />}
            {!!error.message && error.message}
            {images.map(image=>(
                <GifImage key={image.id} url={image.url} title={image.title} />
            ))}
        </>
    )
}