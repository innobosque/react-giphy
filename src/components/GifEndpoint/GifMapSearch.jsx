import { GifItem } from "./GifItem";

export const GifMapSearch = ({ categories }) => {

    return (
        <section>
            <h2 className="text-center text-2xl font-bold">{!!categories.length && 'Resultados'}</h2>
            {
                categories.map(category => (
                    <div className="grid grid-cols-5 gap-5" key={category} >
                        <h3 className="col-span-5 p-5 uppercase">{category}</h3>
                        <GifItem category={category} />
                    </div>
                ))
            }

        </section>
    )
}