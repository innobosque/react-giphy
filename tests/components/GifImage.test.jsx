import { render, screen } from '@testing-library/react';
import { GifImage } from '../../src/components/GifEndpoint/GifImage';

describe(
    'Componente <GifImage />',
    () => {

        const p_url = 'http://placekitten.com/200/300';
        const p_title = 'Texto alternativo';

        test(
            'Debe hacer match con el Snapshot',
            () => {
                const component = render(<GifImage url={p_url} title={p_title} />);
                //El container tiene como una captura
                const { container } = component;
                expect(container).toMatchSnapshot();
                //screen.debug();
            }
        );

        test(
            'Se debe mostrar la imagen con las etiquetas src y alt',
            () => {
                render(<GifImage url={p_url} title={p_title} />);
                screen.debug();
                const { alt, src } = screen.getByRole('img');
                expect(p_title).toBe(alt);
                expect(p_url).toBe(src);
            }
        )

    }
);