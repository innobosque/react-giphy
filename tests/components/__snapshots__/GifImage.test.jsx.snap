// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente <GifImage /> Debe hacer match con el Snapshot 1`] = `
<div>
  <img
    alt="Texto alternativo"
    class="w-full h-full object-cover"
    src="http://placekitten.com/200/300"
  />
</div>
`;
